core/src/sumeragi/network_topology.rs

```diff
@@ -163,38 +159,31 @@ impl Builder {
         self
     }
 
-    /// Set number of view changes after the latest committed block. Default: 0
-    pub fn with_view_changes(mut self, view_change_proofs: ViewChangeProofs) -> Self {
-        self.view_change_proofs = view_change_proofs;
-        self
-    }
-
     /// Build and get topology.
     ///
     /// # Errors
     /// 1. Required field is omitted.
     /// 2. No peer exists.
-    pub fn build(self) -> Result<Topology> {
+    pub fn build(self, number_of_view_changes_this_round: u64) -> Result<Topology> {
```

Comment:
When you do something like this, you're emphasising that the information is important to the network topology. Keep it up.

Reply:
You say "every time" like our jobs are mostly about typing. You want to write the code in such a way that maximises the performance of engineers who have to work on this code later. Forcing the concept that view-changes and topology are linked concepts is therefore good.

Reply:
I'm not really get your point.